homeassistant/components/switcher_kis/__init__.py

```diff
@@ -0,0 +1,93 @@
+"""Home Assistant Switcher Component."""
+
+from asyncio import QueueEmpty, TimeoutError as Asyncio_TimeoutError, wait_for
+from datetime import datetime, timedelta
+from logging import getLogger
+from typing import Dict, Optional
+
+import voluptuous as vol
+
+from homeassistant.components.switch import DOMAIN as SWITCH_DOMAIN
+from homeassistant.const import EVENT_HOMEASSISTANT_STOP
+from homeassistant.core import callback
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.discovery import async_load_platform
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+from homeassistant.helpers.event import async_track_time_interval
+from homeassistant.helpers.typing import EventType, HomeAssistantType
+
+_LOGGER = getLogger(__name__)
+
+DOMAIN = 'switcher_kis'
+
+CONF_DEVICE_ID = 'device_id'
+CONF_DEVICE_PASSWORD = 'device_password'
+CONF_PHONE_ID = 'phone_id'
+
+DATA_DEVICE = 'device'
+
+SIGNAL_SWITCHER_DEVICE_UPDATE = 'switcher_device_update'
+
+ATTR_AUTO_OFF_SET = 'auto_off_set'
+ATTR_ELECTRIC_CURRENT = 'electric_current'
+ATTR_REMAINING_TIME = 'remaining_time'
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_PHONE_ID): cv.string,
+        vol.Required(CONF_DEVICE_ID): cv.string,
+        vol.Required(CONF_DEVICE_PASSWORD): cv.string
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+
+async def async_setup(hass: HomeAssistantType, config: Dict) -> bool:
+    """Set up the switcher component."""
+    from aioswitcher.bridge import SwitcherV2Bridge
+
+    phone_id = config[DOMAIN][CONF_PHONE_ID]
+    device_id = config[DOMAIN][CONF_DEVICE_ID]
+    device_password = config[DOMAIN][CONF_DEVICE_PASSWORD]
+
+    v2bridge = SwitcherV2Bridge(
+        hass.loop, phone_id, device_id, device_password)
+
+    await v2bridge.start()
+
+    async def async_stop_bridge(event: EventType) -> None:
+        """On homeassistant stop, gracefully stop the bridge if running."""
+        await v2bridge.stop()
+
+    hass.async_add_job(hass.bus.async_listen_once(
```

Comment:
We shouldn't use `hass.async_add_job` anymore. It's legacy.

Why do we need to add a job for this at all?

Reply:
@MartinHjelmare 
Much appreciated!
I addressed this and the other issues you referenced in a new PR I'm currently working on.
I'll notify you once I'll create the new pull request.

Thank you very much.
:-)

Reply:
The new PR is [here](https://github.com/home-assistant/home-assistant/pull/23477).
Besides resolving all the change requests from this PR, it also adds a couple of services and update the requirement version.
:-)